service/history/replication/executable_task.go

```diff
@@ -107,6 +107,7 @@ type (
 		// mutable data
 		taskState int32
 		attempt   int32
+		namespace atomic.Value
```

Comment:
why do we need it instead of load it from ns cache?

Reply:
we can get the ns name from cache when emitting the metric, but usually task will be executed (within execution, ns cache will be accessed) before acking / nacking; 
so, less cache access